Assignment4/JAVSIE-y160338/task07.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1VDY_dFtAOMPFqC4DuLAi4BKdEgBnpa9a
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+# Visualize the results
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf ns:Person
+  }
+  ''',
+  initNs = { "ns": Namespace("http://somewhere#"), "rdfs": RDFS}
+)
+
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+# Visualize the results
+
+q1 = prepareQuery('''
+  SELECT DISTINCT ?Subject WHERE { 
+    {?Subject rdf:type ns:Person. }
+    UNION 
+    {?s rdfs:subClassOf ns:Person.
+    ?Subject rdf:type ?x}
```

Comment:
this is wrong. It should be `?Subject rdf:type ?s ` otherwise you are asking for all subclasses of any class, not just Person